pipelines/controller/src/main/java/com/google/fhir/analytics/PipelineManager.java

```diff
@@ -554,6 +554,8 @@ public void run() {
         manager.captureError(currentDwhRoot, e);
         manager.setLastRunDetails(currentDwhRoot, FAILURE);
         manager.setLastRunStatus(LastRunStatus.FAILURE);
+      } finally {
+        System.gc();
```

Comment:
Review to myself: Add a comment why this is needed.

Reply:
Can we move this change over [here](https://github.com/google/fhir-data-pipes/blob/c6cebd45d3138499c0bd0e1a318a0046ea6a6dc7/pipelines/batch/src/main/java/com/google/fhir/analytics/EtlUtils.java#L89C10-L89C10) so that this is triggered after each pipeline run, because in case of incremental run we trigger pipeline run twice.

Reply:
That's a fair point but I think it is fine to trigger GC only once here; please note this is only a hint to the compiler and we cannot _force_ a GC. In fact, it usually takes a little bit of time for memory usage to go down after each pipeline run. On the other hand if GC is triggered between the two pipelines (incremental and merge) it can add some lag, I think.